File: LeetCode #1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        long long int len=nums.size();
        vector<int> res;
        bool x=false;
        for(int i=0 ; i<len ; i++) {
            for(int j=i+1 ; j<len ; j++) {
                if(nums[i]+nums[j] == target) {
                    res.push_back(i);
                    res.push_back(j);
                    x=true;
                    break;
                }
            }
            if(x)
                break;
        }
        return res;
    }
};

File: LeetCode #1.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;

        HashMap<Integer, Integer> hmap = new HashMap<>();

        int[] ans = new int[2];

        for (int i = 0; i < n; i++) {
            int key = target - nums[i];

            if (hmap.containsKey(key)) {
                ans[0] = hmap.get(key);
                ans[1] = i;
                break;
            }

            hmap.put(nums[i], i);
        }

        return ans;
    }
}